### PR TITLE
Fix smooth scroll android

### PR DESCRIPTION
**Pull request by Latropos**
## Summary
Fix this issue: https://github.com/software-mansion/react-native-reanimated/issues/4200
It looks that smoothScrollTo was not always working on Android. According to this stackOverflow [question](https://stackoverflow.com/questions/18673065/scrollto-always-works-smoothscrollto-only-sometimes) we should post a runnable here.

## Test plan

Tested 4 cases in our example App: Horizontal and Vertical Scrolls with both animated and instant transitions. 

<table>
<tr><td>Animated</td><td>Not Animated</td></tr>
<tr><td>

https://user-images.githubusercontent.com/56199675/226555227-3b75ba0b-2861-47ea-955f-56e8c368ad0f.mov

</td><td>

https://user-images.githubusercontent.com/56199675/226555278-6c0f329a-4417-4271-a337-bd28b41c28f2.mov

</td></tr>

<tr><td>

https://user-images.githubusercontent.com/56199675/226556289-8b527a54-8f8e-4b4d-9ff5-5ce4e05414d3.mov

</td><td>

https://user-images.githubusercontent.com/56199675/226556339-2dbe1e56-05e5-4923-8f4d-80050ade8bbe.mov

</td></tr>

</table>

### CODE REVIEW

File: android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java [hunk 1]
```diff
@@ -57,9 +57,22 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
 
     if (animated) {
       if (horizontal) {
-        ((ReactHorizontalScrollView) view).smoothScrollTo((int) x, (int) y);
+        View finalView = view;
+        view.post(new Runnable() {
+          @Override
+          public void run() {
+            ((ReactHorizontalScrollView) finalView).smoothScrollTo(x, y);
```
Comment (piaskowyk) [inline android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java:64]: The `finalView` should be `ReactHorizontalScrollView`, then we will get rid off the casting here.

File: android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java [hunk 2]
```diff
@@ -57,9 +57,22 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
...
+          }
+        });
+
       } else {
-        ((ReactScrollView) view).smoothScrollTo((int) x, (int) y);
+        View finalView = view;
+        view.post(new Runnable() {
+          @Override
+          public void run() {
+            ((ReactScrollView) finalView).smoothScrollTo(x, y);
```
Comment (piaskowyk) [inline android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java:73]: Same as above

**Comment by piaskowyk** [comment]
To fix CI you need to run: `cd android && ./gradlew :spotlessApply`

File: android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java [hunk 3]
```diff
@@ -57,9 +57,24 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
...
       if (horizontal) {
-        ((ReactHorizontalScrollView) view).smoothScrollTo((int) x, (int) y);
+        ReactHorizontalScrollView finalView = view;
+        view.post(
+            new Runnable() {
+              @Override
+              public void run() {
+                finalView.smoothScrollTo(x, y);
+              }
+            });
```
Comment (piaskowyk) [inline android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java:67]: It looks like code duplication with lines 71-77. Could we move it outside the `if` condition?

Comment (Latropos) [reply android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java:67]: I think we can't, we have to call view.post in the same context we define the view variable.

File: android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java [hunk 4]
```diff
@@ -56,17 +58,17 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
...
+                  }
+
+              });
     } else {
-      if (horizontal) {
-        ((ReactHorizontalScrollView) view).scrollTo((int) x, (int) y);
-      } else {
-        ((ReactScrollView) view).scrollTo((int) x, (int) y);
-      }
+        (view).scrollTo((int) x, (int) y);
```
Comment (tomekzaw) [inline android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java:71]:
```suggestion
        view.scrollTo(x, y);
```

**Review by piaskowyk — CHANGES_REQUESTED** [review]
We no longer need this variable - https://github.com/software-mansion/react-native-reanimated/pull/4238/files#diff-ceb86f6a766924630c94e88c28e404931dee8fdc3c576ac0461f440b164552e0R45

File: android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java [hunk 5]
```diff
@@ -41,9 +41,7 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
     int y = Math.round(PixelUtil.toPixelFromDIP(argY));
     boolean horizontal = false;
```
Comment (piaskowyk) [inline android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java:42]: I still see it 😄

**Review by piaskowyk — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java [hunk 6]
```diff
@@ -56,17 +53,21 @@ public static void scrollTo(View view, double argX, double argY, boolean animate
...
+              }
+            }
+          });
     } else {
-      if (horizontal) {
-        ((ReactHorizontalScrollView) view).scrollTo((int) x, (int) y);
-      } else {
-        ((ReactScrollView) view).scrollTo((int) x, (int) y);
-      }
+      (view).scrollTo((int) x, (int) y);
```
Comment (piaskowyk) [inline android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java:70]:
```suggestion
      view.scrollTo(x, y);
```

**Review by piaskowyk — CHANGES_REQUESTED** [review]

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**